plutus-chain-index-core/src/Plutus/ChainIndex/Emulator/Handlers.hs

```diff
@@ -187,11 +192,14 @@ appendBlocks blocks = do
                     logDebug $ InsertionSuccess tip_ insertPosition
                     return (newIndex, transactions ++ txs)
     oldState <- get @ChainIndexEmulatorState
+    let oldIndex = view utxoIndex oldState
     (newIndex, transactions) <- foldM processBlock (view utxoIndex oldState, []) blocks
-    put $ oldState
-            & set utxoIndex newIndex
-            & over diskState
-                (mappend $ foldMap (\(tx, opt) -> if tpoStoreTx opt then DiskState.fromTx tx else mempty) transactions)
+    -- This check reduces the emulator's memory consumption by 75%
```

Comment:
Is it valid to not store transactions that don't have an effect on the index?

Reply:
I have checked to make sure that we don't miss anything and turned out this is the case when there are no transactions at all. But I suppose adding an if-branch to validate it in code would be a good idea.